Spec conventions (#119)
tc39/how-we-work: spec-conventions.md

# Conventions when writing specification text

> **NOTE:** This document is currently a skeleton, and should not be considered a
> complete list.
> The issue [tc39/ecma262#2236](https://github.com/tc39/ecma262/issues/2236) is
> a good place to discuss conventions.

## Spelling, grammar, and typography

## Notation

## Algorithm steps

## Abstract operations

## Implementations of functions and methods

### Order of observable operations

When writing the algorithm steps for the implementation of a JS built-in
function or method, this order should be followed:

> 1. Process and validate each argument, in order.
> 2. Perform validation of any other preconditions, if any.
> 3. Perform the actual work of the function.

Reviewer: ```suggestion
1. If applicable, process and validate the receiver.
2. Process and validate each argument, in order.
3. Perform validation of any other preconditions, if any.
4. Perform the actual work of the function.
```

Reviewer 2: ```suggestion
1. If applicable, process and validate the receiver.
1. Process and validate each argument, in order.
1. Perform validation of any other preconditions, if any.
1. Perform the actual work of the function.
```
nit :-)

Reviewer: I don't care, the first is easier to _read_ in source form while the second is more convenient to _edit_, and both render identically.

Reviewer 2: that's why it's a nit. the main benefit of the latter is reducing diff churn when you add or remove an item.

# Conventions when writing specification text

> **NOTE:** This document is currently a skeleton, and should not be considered a
> complete list.
> The issue [tc39/ecma262#2236](https://github.com/tc39/ecma262/issues/2236) is
> a good place to discuss conventions.

> ## Spelling, grammar, and typography
> 

Reviewer: ```suggestion
## Spelling, grammar, and typography

Specification text should use Oxford spelling (e.g., "behaviour" rather than "behavior").
```

Reviewer 3: No, this is an editorial convention, which we will document on the ecma262 wiki soon™️. See https://github.com/tc39/how-we-work/pull/119#issuecomment-1326796200
